ensemble/randomforest.h

```diff
@@ -0,0 +1,53 @@
+#ifndef _MACHINELEARNING_ENSEMBLE_RANDOMFOREST_H_
+#define _MACHINELEARNING_ENSEMBLE_RANDOMFOREST_H_
+
+#include "../tree/tree.h"
+
+namespace ensemble {
+/**
+ * Base class for Random Forest Classification and Regression algorithms.
+ */
+class RandomForest {
+ public:
+  /**
+   * \brief Standard destructor.
+   */
+  virtual ~RandomForest();
+
+  /**
+   * Compute and return the score of the training dataset obtained using an
+   * out-of-bag estimate
+   * \return oob score
+   */
+  virtual double OobScore() = 0;
+
+  /*
+   * Build the RandomForest
+   * \param thread Number of threads for paralel execution
+   * \param tree Number of trees in the Forest
+   */
+  virtual void Build(/* Collection col, */ const int thread,
+                     const int tree) = 0;
+
+  /**
+   * Predicts the labels of the instances of X based on a vote by
```

Comment:
I think that for this function we should provide the brief and a larger comment explaining how this prediction is made. Also, since this is an interface, not all random forests are based on the vote (which I am assuming here that you meant majority vote). Please Provide a better explanation of the tasks that this function will do.